src/node_sea.cc

```diff
@@ -194,51 +231,47 @@ std::optional<SeaConfig> ParseSingleExecutableConfig(
   return result;
 }
 
-bool GenerateSingleExecutableBlob(const SeaConfig& config) {
+ExitCode GenerateSingleExecutableBlob(const SeaConfig& config) {
   std::string main_script;
   // TODO(joyeecheung): unify the file utils.
   int r = ReadFileSync(&main_script, config.main_path.c_str());
   if (r != 0) {
     const char* err = uv_strerror(r);
     FPrintF(stderr, "Cannot read main script %s:%s\n", config.main_path, err);
-    return false;
+    return ExitCode::kGenericUserError;
   }
 
-  std::vector<char> sink;
-  // TODO(joyeecheung): reuse the SnapshotSerializerDeserializer for this.
-  sink.reserve(SeaResource::kHeaderSize + main_script.size());
-  const char* pos = reinterpret_cast<const char*>(&kMagic);
-  sink.insert(sink.end(), pos, pos + sizeof(kMagic));
-  pos = reinterpret_cast<const char*>(&(config.flags));
-  sink.insert(sink.end(), pos, pos + sizeof(SeaFlags));
-  sink.insert(
-      sink.end(), main_script.data(), main_script.data() + main_script.size());
-
-  uv_buf_t buf = uv_buf_init(sink.data(), sink.size());
+  SeaResource sea{config.flags,
+                  std::string_view{main_script.data(), main_script.size()}};
```

Comment:
```suggestion
  SeaResource sea{config.flags, main_script};
```

or `std::string_view{main_script}` if you prefer the explicit ctor